src/test/client/nonblocking.cc

```diff
@@ -611,3 +612,499 @@ TEST_F(TestClient, LlreadvLlwritevInvalidFileHandle) {
   ASSERT_EQ(bytes_read, -CEPHFS_EBADF);
   ASSERT_EQ(bl.length(), 0);
 }
+
+TEST_F(TestClient, LlreadvLlwritevOverlimit) {
```

Comment:
/a/vshankar-2024-08-01_09:58:56-fs-wip-vshankar-testing-20240801.064407-debug-testing-default-smithi/7829908

I see the following crash

```
INFO:tasks.workunit.client.0.smithi142.stderr:    -1> 2024-08-02T07:09:20.600+0000 7f987f000640 20 client.4689 put_inode on 0x1.head(faked_ino=0 nref=9 ll_ref=2 cap_refs={} open={} mode=40755 size=0/0 nlink=1 btime=2024-08-02T07:08:01.737249+0000 mtime=2024-08-02T07:09:20.594231+0000 ctime=2024-08-02T07:09:20.594231+0000 change_attr=18 caps=pAsLsXs(0=pAsLsXs) has_dir_layout 0x7f986000c330) n = 1
2024-08-02T07:09:20.717 INFO:tasks.workunit.client.0.smithi142.stderr:     0> 2024-08-02T07:09:20.601+0000 7f988880c480 -1 *** Caught signal (Aborted) **
2024-08-02T07:09:20.717 INFO:tasks.workunit.client.0.smithi142.stderr: in thread 7f988880c480 thread_name:ceph_test_clien
2024-08-02T07:09:20.717 INFO:tasks.workunit.client.0.smithi142.stderr:
2024-08-02T07:09:20.717 INFO:tasks.workunit.client.0.smithi142.stderr: ceph version 19.0.0-5453-g5f7f38f6 (5f7f38f6943da1b8f6e6fe9aed0202920c629e98) squid (dev)
2024-08-02T07:09:20.717 INFO:tasks.workunit.client.0.smithi142.stderr: 1: /lib64/libc.so.6(+0x3e6f0) [0x7f988863e6f0]
2024-08-02T07:09:20.717 INFO:tasks.workunit.client.0.smithi142.stderr: 2: /lib64/libc.so.6(+0x8b94c) [0x7f988868b94c]
2024-08-02T07:09:20.717 INFO:tasks.workunit.client.0.smithi142.stderr: 3: raise()
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 4: abort()
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 5: /lib64/libc.so.6(+0x29130) [0x7f9888629130]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 6: /lib64/libc.so.6(+0x11d77b) [0x7f988871d77b]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 7: /lib64/libc.so.6(+0x11c0e6) [0x7f988871c0e6]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 8: __strcat_chk()
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 9: (TestClient_LlreadvLlwritevOverlimit_Test::TestBody()+0x2c4) [0x55e475d8bd5c]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 10: ceph_test_client(+0x1fee96) [0x55e475ec2e96]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 11: ceph_test_client(+0x204b8c) [0x55e475ec8b8c]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 12: (testing::Test::Run()+0xb8) [0x55e475ea60a2]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 13: (testing::TestInfo::Run()+0x108) [0x55e475ea61ac]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 14: (testing::TestSuite::Run()+0xb6) [0x55e475ea6264]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 15: (testing::internal::UnitTestImpl::RunAllTests()+0x2dd) [0x55e475eb8be5]
2024-08-02T07:09:20.718 INFO:tasks.workunit.client.0.smithi142.stderr: 16: ceph_test_client(+0x1feef6) [0x55e475ec2ef6]
2024-08-02T07:09:20.719 INFO:tasks.workunit.client.0.smithi142.stderr: 17: ceph_test_client(+0x205026) [0x55e475ec9026]
2024-08-02T07:09:20.719 INFO:tasks.workunit.client.0.smithi142.stderr: 18: (testing::UnitTest::Run()+0x95) [0x55e475eb48a3]
2024-08-02T07:09:20.719 INFO:tasks.workunit.client.0.smithi142.stderr: 19: main()
2024-08-02T07:09:20.719 INFO:tasks.workunit.client.0.smithi142.stderr: 20: /lib64/libc.so.6(+0x29590) [0x7f9888629590]
2024-08-02T07:09:20.719 INFO:tasks.workunit.client.0.smithi142.stderr: 21: __libc_start_main()
2024-08-02T07:09:20.719 INFO:tasks.workunit.client.0.smithi142.stderr: 22: _start()
```

Reply:
it's complaining about string concatenation, should be this line `strcat(out_str, num);` in `TEST_F(TestClient, LlreadvLlwritevOverlimit)`.

As per https://refspecs.linuxbase.org/LSB_4.1.0/LSB-Core-generic/LSB-Core-generic/libc---strcat-chk-1.html: `__strcat_chk() shall check for buffer overflow before computing a result. If an overflow is anticipated, the function shall abort and the program calling it shall exit.`

Looks like there was buffer overflow

Reply:
@vshankar [simplied](https://github.com/ceph/ceph/compare/1b844d412baa556df66de1e7f7e7482be482b3d4..7c42357516d4fcdae8877cd749904344c2d56691) the test case. This shouldn't bother now.